src/crimson/os/seastore/seastore_types.h

```diff
@@ -62,55 +65,46 @@ struct segment_id_t {
     0xF << (std::numeric_limits<internal_segment_id_t>::digits - DEVICE_ID_LEN_BITS);
   // default internal segment id
   static constexpr internal_segment_id_t DEFAULT_INTERNAL_SEG_ID =
-    std::numeric_limits<internal_segment_id_t>::max() - 1;
+    (std::numeric_limits<internal_segment_id_t>::max() >> DEVICE_ID_LEN_BITS) - 1;
 
   internal_segment_id_t segment = DEFAULT_INTERNAL_SEG_ID;
 
-  constexpr segment_id_t(uint32_t encoded) : segment(encoded) {}
 public:
+  //constexpr segment_id_t(uint32_t encoded) : segment(encoded) {}
   constexpr static segment_id_t make_max() {
-    return std::numeric_limits<internal_segment_id_t>::max();
+    return std::numeric_limits<internal_segment_id_t>::max() >> DEVICE_ID_LEN_BITS;
   }
   constexpr static segment_id_t make_null() {
-    return std::numeric_limits<internal_segment_id_t>::max() - 1;
+    return (std::numeric_limits<internal_segment_id_t>::max() >> DEVICE_ID_LEN_BITS) - 1;
   }
   /* Used to denote relative paddr_t */
   constexpr static segment_id_t make_record_relative() {
-    return std::numeric_limits<internal_segment_id_t>::max() - 2;
+    return (std::numeric_limits<internal_segment_id_t>::max() >> DEVICE_ID_LEN_BITS) - 2;
   }
   constexpr static segment_id_t make_block_relative() {
-    return std::numeric_limits<internal_segment_id_t>::max() - 3;
+    return (std::numeric_limits<internal_segment_id_t>::max() >> DEVICE_ID_LEN_BITS) - 3;
   }
   // for tests which generate fake paddrs
   constexpr static segment_id_t make_fake() {
-    return std::numeric_limits<internal_segment_id_t>::max() - 4;
+    return (std::numeric_limits<internal_segment_id_t>::max() >> DEVICE_ID_LEN_BITS) - 4;
   }
 
   /* Used to denote references to notional zero filled segment, mainly
    * in denoting reserved laddr ranges for unallocated object data.
    */
   constexpr static segment_id_t make_zero() {
-    return std::numeric_limits<internal_segment_id_t>::max() - 5;
+    return (std::numeric_limits<internal_segment_id_t>::max() >> DEVICE_ID_LEN_BITS) - 5;
   }
   constexpr static segment_id_t make_delayed() {
-    return std::numeric_limits<internal_segment_id_t>::max() - 6;
+    return (std::numeric_limits<internal_segment_id_t>::max() >> DEVICE_ID_LEN_BITS) - 6;
   }
 
   segment_id_t() = default;
-  segment_id_t(device_id_t id, device_segment_id_t segment)
-    : segment(make_internal(segment, id)) {
-    // only lower 4 bits are effective, and we have to reserve 0x0F for
-    // special XXX_SEG_IDs
-    assert(id < DEVICE_ID_MAX);
-  }
-
-  [[gnu::always_inline]]
-  device_id_t device_id() const {
-    return internal_to_device(segment);
-  }
+  constexpr segment_id_t(device_segment_id_t segment)
+    : segment(segment) {}
 
   [[gnu::always_inline]]
-  device_segment_id_t device_segment_id() const {
+  device_segment_id_t segment_id() const {
     return internal_to_segment(segment);
   }
```

Comment:
This portion strips the device id out of segment_id_t.  I don't really see a reason to do that, a bunch of interfaces are already set up to take a segment_id_t that includes the device id.